components/core/src/service.rs

```diff
@@ -83,7 +81,89 @@ impl FromStr for BindingMode {
     }
 }
 
-////////////////////////////////////////////////////////////////////////
+/// A binding from a service name to a service group that provides that service
+#[derive(Clone, Debug, Eq, Hash, PartialEq)]
+pub struct ServiceBind {
+    name: String,
+    service_group: ServiceGroup,
+}
+
+impl ServiceBind {
+    pub fn new<S: Into<ServiceGroup>>(name: &str, service_group: S) -> Self {
+        Self {
+            name: name.to_string(),
+            service_group: service_group.into(),
+        }
+    }
+
+    pub fn name(&self) -> &String {
+        &self.name
+    }
+
+    pub fn service_group(&self) -> &ServiceGroup {
+        &self.service_group
+    }
+}
+
+impl FromStr for ServiceBind {
+    type Err = Error;
+
+    fn from_str(bind_str: &str) -> result::Result<Self, Self::Err> {
+        match bind_str.split(':').collect::<Vec<_>>().as_slice() {
+            [name, sg_str] => {
+                ServiceGroup::from_str(sg_str).and_then(|sg| Ok(ServiceBind::new(name, sg)))
```

Comment:
```suggestion
                ServiceGroup::from_str(sg_str)..map(|sg| ServiceBind::new(name, sg))
```
In general, if the body of an `and_then` always returns `Ok(…)` (or `Some(…)` if the receiver is an `Option`), it can be replaced with a simpler `map`.